.github/workflows/build.yml

```diff
@@ -2,14 +2,65 @@ name: Pipeline
 
 on: [push, pull_request]
 
+env:
+  GRADLE_OPTS: "-Dorg.gradle.internal.launcher.welcomeMessageEnabled=false"
+
+concurrency:
+  cancel-in-progress: true
+  group: build-${{ github.event.pull_request.number || github.event.after }}
```

Comment:
This allows only one workflow run per PR and, in the meantime, does not affect the push event (because `github.event.after` will be unique for every commit). @charleskorn does it look okay?

Reply:
LGTM, thanks.